pkg/network/usm/ebpf_gotls.go

```diff
@@ -178,14 +146,56 @@ func newGoTLSProgramProtocolFactory(m *manager.Manager) protocols.ProtocolFactor
 			return nil, errors.New("goTLS support requires runtime-compilation or CO-RE to be enabled")
 		}
 
+		attacherCfg := uprobes.AttacherConfig{
+			EbpfConfig: &c.Config,
+			Rules: []*uprobes.AttachRule{{
+				Targets: uprobes.AttachToExecutable,
+				ProbesSelector: []manager.ProbesSelector{
+					&manager.AllOf{
+						Selectors: []manager.ProbesSelector{
+							&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{EBPFFuncName: connReadProbe}},
+							&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{EBPFFuncName: connReadRetProbe}},
+							&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{EBPFFuncName: connWriteProbe}},
+							&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{EBPFFuncName: connWriteRetProbe}},
+							&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{EBPFFuncName: connCloseProbe}},
+						},
+					},
+				},
+				ProbeOptionsOverride: map[string]uprobes.ProbeOptions{
+					connReadProbe:     {IsManualReturn: false, Symbol: bininspect.ReadGoTLSFunc},
+					connReadRetProbe:  {IsManualReturn: true, Symbol: bininspect.ReadGoTLSFunc},
+					connWriteProbe:    {IsManualReturn: false, Symbol: bininspect.WriteGoTLSFunc},
+					connWriteRetProbe: {IsManualReturn: true, Symbol: bininspect.WriteGoTLSFunc},
+					connCloseProbe:    {IsManualReturn: false, Symbol: bininspect.CloseGoTLSFunc},
+				},
+			}},
+			ExcludeTargets:                 uprobes.ExcludeInternal,
+			PerformInitialScan:             false,
+			EnablePeriodicScanNewProcesses: false,
```

Comment:
@gjulianm why do those values set to false?

Reply:
The previous GoTLS attacher had the same behavior, it didn't have an initial scan of all processes nor it checked periodically for processes that we hadn't considered due to missed events. It can be changed as needed.

Reply:
Commenting here for the record: the GoTLS attacher relies on the process monitor, which does a scan of all processes once its `Initialize` method is called.